library/alloc/src/vec/mod.rs

```diff
@@ -1001,12 +1001,14 @@ impl<T, A: Allocator> Vec<T, A> {
     /// If `len` is greater than the vector's current length, this has no
     /// effect.
     ///
-    /// The [`drain`] method can emulate `truncate`, but causes the excess
-    /// elements to be returned instead of dropped.
-    ///
     /// Note that this method has no effect on the allocated capacity
     /// of the vector.
     ///
+    /// The [`split_off`] method is similar to `truncate`, but causes the
+    /// tail elements to be returned as a newly-allocated `Vec` instead
+    /// of being dropped. Note that [`Vec::drain`] can emulate `split_off`
+    /// and avoids this extra allocation by returning an iterator.
+    ///
```

Comment:
The dance "`split_off` emulates `truncate` and `drain` emulates `split_off`" seems a bit weird. I would just mention that `Vec::drain` also returns the tail elements, but as an iterator rather than in a new allocation.

I would mention the `panic` thing here. I saw you put it later, but I don't think that's the right place. The issue with panic is not that `truncate` doesn't panic, rather that `split_off` and `drain` can emulate `truncate` only for indexes not bigger than the length, otherwise they will panic. So I think the right place to mention it should be here, probably after mentioning `Vec::drain`.